url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/event/net/mock/KyoroDatagramMock.java

```diff
@@ -0,0 +1,294 @@
+package net.hetimatan.util.event.net.mock;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.nio.channels.SelectableChannel;
+import java.util.LinkedList;
+import java.util.Stack;
+
+
+import net.hetimatan.util.event.net.io.KyoroDatagram;
+import net.hetimatan.util.event.net.io.KyoroSelector;
+import net.hetimatan.util.http.HttpObject;
+import net.hetimatan.util.io.ByteArrayBuilder;
+
+//
+// test用
+// 画面上に表示する
+//
+public class KyoroDatagramMock extends KyoroDatagram {
+	public static final int NAT_TYPE_OPEN_INTERNET = 0;
+	public static final int NAT_TYPE_FULL_CONE = 1;
+	public static final int NAT_TYPE_RESTRICTED_ADDRESS = 2;
+	public static final int NAT_TYPE_RESTRICTED_PORT = 3;
+	public static final int NAT_TYPE_SYMMETRIC = 4;
+
+	private int mNatType = 0;
+	// for debug
+	private int mDeviceId = 0;
+	public KyoroDatagramMock(int type, int id) {
+		mNatType = type;
+		mDeviceId = id;
+	}
+
+	public int getDeviceId() {
+		return mDeviceId;
+	}
+
+	public int getNatType() {
+		return mNatType;
+	}
+
+	private Stack<DatagramPacket> mPackets = new Stack<>();
+	private WeakReference<KyoroSelector> mCurrentSelector = null;
+	public void onReceivePacket(byte[] content, byte[] ip) throws IOException {
+		if(!isConnectable(ip)) {
+			//throw new IOException("failed connect");
+			return;
+		}
+		mPackets.push(new DatagramPacket(content, ip));
+		if(mCurrentSelector == null) {
+			return;
+		}
+		KyoroSelector selector = this.mCurrentSelector.get();
+		if(selector == null) {
+			return;
+		}
+		selector.wakeup(this, KyoroSelector.READ);
+	}
+
+	public boolean isConnectable(byte[] ip) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.